src/ln/channelmanager.rs

```diff
@@ -2956,7 +2956,21 @@ impl<'a, R : ::std::io::Read> ReadableArgs<R, ChannelManagerReadArgs<'a>> for (S
 			}
 		}
 
+		let mut pending_events = Vec::new();
+
 		let forward_htlcs_count: u64 = Readable::read(reader)?;
+		let next_forward = if forward_htlcs_count == 0 {
+			Instant::now()
+		} else {
+			// Wait five seconds to forward things we got off disk to give them a chance to
+			// reconnect to their peers. Probably all the HTLCs have timed out by now, but try
+			// anyway.
+			let now_plus_five = Instant::now() + Duration::from_secs(5);
```

Comment:
the five seconds should probably be put in a constant. Alternatively, the peer reconnections could be awaited.